test/Constraints/crash-85843677.swift

```diff
@@ -0,0 +1,15 @@
+// RUN: %target-typecheck-verify-swift
```

Comment:
One more nit - you'd have to add `not` I think here since I have landed my revert so this would no longer crash.